server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -689,18 +689,28 @@ private void resourceLimitCheck(Account owner, Boolean displayVm, Long cpu, Long
     }
 
     protected void resourceCountIncrement(long accountId, Boolean displayVm, Long cpu, Long memory) {
-        if (! VirtualMachineManager.ResourceCountRunningVMsonly.value()) {
-            _resourceLimitMgr.incrementResourceCount(accountId, ResourceType.user_vm, displayVm);
-            _resourceLimitMgr.incrementResourceCount(accountId, ResourceType.cpu, displayVm, cpu);
-            _resourceLimitMgr.incrementResourceCount(accountId, ResourceType.memory, displayVm, memory);
+        if (!VirtualMachineManager.ResourceCountRunningVMsonly.value()) {
+            Transaction.execute(new TransactionCallbackNoReturn() {
```

Comment:
Another way to do this without incrementing all three in a transaction would require adding columns for each reservation type (`user_vm`, `cpu`, `memory`) in `user_vm` table since `cpu` & `memory` are calculated using the `user_vm` table.

Reply:
Similar solution for primary_storage & volume as well.